frameworklauncher/src/main/java/com/microsoft/frameworklauncher/client/LauncherClient.java

```diff
@@ -184,6 +185,16 @@ public void putClusterConfiguration(ClusterConfiguration clusterConfiguration) t
     });
   }
 
+  public void putAclConfiguration(AclConfiguration aclConfiguration) throws Exception {
+    executeWithRetry(() -> {
+      CommonValidation.validate(aclConfiguration);
```

Comment:
>CommonValidation.validate(aclConfiguration); [](start = 6, length = 44)

Is validate a local action? If yes, it is reliable and can move out of executeWithRetry() so that it will not waste retries on validations. Also exception of validations can be thrown to caller directly to gives out clear error stack if moved out.

Reply:
For LauncherClient, it only throws LauncherClientException. This helps to unify validate exception from remote and local to the same interface for user to handle. For example, here, the bad request exception and 400 will be wrapped into LauncherClientException, the same as from remote exception. BTW, local validate is not a performance bottleneck here.

---
In reply to: [167463665](https://github.com/Microsoft/pai/pull/320#discussion_r167463665) [](ancestors = 167463665)